inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -58,20 +60,45 @@ export class CanvasGraphService {
         this.clear();
 
         // Get global min max of time axis (across all datasets).
-        let globalTimeMinMax = {min: Infinity, max: 0};
+        const globalTimeMinMax = {min: Infinity, max: 0};
 
-        // TODO: Make better sliding window code (accounting for zoom and pan).
         // TODO: Perhaps see if i can reduce the number of allocations.
         // Keep only visible and non empty datasets and get a certain window of items.
-        const datasets = this.datasets.filter((dataset: IPerfDataset) => !dataset.hidden && dataset.data.length > 0).map((dataset: IPerfDataset) => ({
-            ...dataset,
-            data: dataset.data.slice(Math.max(dataset.data.length - this._sizeOfWindow, 0))
-        }));
+        const datasets = this.datasets.map((dataset: IPerfDataset) => {
+            // skip hidden and empty datasets!
+            if (dataset.data.length === 0 || !!dataset.hidden) {

```

Comment:
thinking through overallocation in this whole section will be necessary :-)

Reply:
Yep, of course!! I have it pinned as a to-do item 😄. Probably going to have a typed array hold the items, as we've discussed. Going to put it off until the data layer is implemented though!!